### ISSUE TITLE
IOT Agent crashing after startup. 

### ISSUE THREAD

**Issue by jdiller** [DataDog/datadog-agent#11464]
**Describe what happened:**
After many months of flawless performance, the agent on a Raspi 4B with Rasbpian 10 started crashing on startup, I think after an `apt-get upgrade` of everything on the OS (there isn't much, I only use it as a stats collector for my solar panels pumped over to DataDog via a python script and the statsd server). 

I've tried removing and reinstalling the agent and also stopping my stats collector script to rule out that it is sending bad data somehow but I can't figure out what to do to keep the agent running. Any help or suggestions would be appreciated. 

**Additional environment details (Operating System, Cloud provider, etc):**
Here are the logs from the crash: 
```
Mar 26 12:25:28 hallowell agent[12907]: 2022-03-26 12:25:28 EDT | CORE | INFO | (pkg/collector/worker/check_logger.go:37 in CheckStarted) | check:disk | Running check...
Mar 26 12:25:28 hallowell agent[12907]: 2022-03-26 12:25:28 EDT | CORE | INFO | (pkg/collector/worker/check_logger.go:56 in CheckFinished) | check:disk | Done running check
Mar 26 12:25:33 hallowell agent[12907]: panic: unaligned 64-bit atomic operation
Mar 26 12:25:33 hallowell agent[12907]: goroutine 191 [running]:
Mar 26 12:25:33 hallowell agent[12907]: runtime/internal/atomic.panicUnaligned()
Mar 26 12:25:33 hallowell agent[12907]: #011/root/.gimme/versions/go1.16.12.linux.arm/src/runtime/internal/atomic/unaligned.go:8 +0x24
Mar 26 12:25:33 hallowell agent[12907]: runtime/internal/atomic.Xadd64(0x4ef08b4, 0x1, 0x0, 0xa6cbc8, 0x4d41500)
Mar 26 12:25:33 hallowell agent[12907]: #011/root/.gimme/versions/go1.16.12.linux.arm/src/runtime/internal/atomic/asm_arm.s:233 +0x14
Mar 26 12:25:33 hallowell agent[12907]: github.com/DataDog/datadog-agent/pkg/metrics.(*IterableSeries).Append(0x4ef08a0, 0x516bda0)
Mar 26 12:25:33 hallowell agent[12907]: #011/omnibus/src/datadog-iot-agent/src/github.com/DataDog/datadog-agent/pkg/metrics/iterable_series.go:46 +0x48
Mar 26 12:25:33 hallowell agent[12907]: github.com/DataDog/datadog-agent/pkg/aggregator.(*TimeSampler).dedupSerieBySerieSignature(0x4be4408, 0x4bf4d50, 0x1, 0x2, 0x2044278, 0x4ef08a0, 0x4d41810)
Mar 26 12:25:33 hallowell agent[12907]: #011/omnibus/src/datadog-iot-agent/src/github.com/DataDog/datadog-agent/pkg/aggregator/time_sampler.go:172 +0x3e8
Mar 26 12:25:33 hallowell agent[12907]: github.com/DataDog/datadog-agent/pkg/aggregator.(*TimeSampler).flushSeries.func1(0x4bf4d50, 0x1, 0x2)
Mar 26 12:25:33 hallowell agent[12907]: #011/omnibus/src/datadog-iot-agent/src/github.com/DataDog/datadog-agent/pkg/aggregator/time_sampler.go:130 +0x4c
Mar 26 12:25:33 hallowell agent[12907]: github.com/DataDog/datadog-agent/pkg/metrics.(*ContextMetricsFlusher).FlushAndClear.func2()
Mar 26 12:25:33 hallowell agent[12907]: #011/omnibus/src/datadog-iot-agent/src/github.com/DataDog/datadog-agent/pkg/metrics/context_metrics_flusher.go:69 +0x3c
Mar 26 12:25:33 hallowell agent[12907]: github.com/DataDog/datadog-agent/pkg/metrics.aggregateContextMetricsByContextKey(0x4bf4d40, 0x1, 0x2, 0x4d41624, 0x4d41618)
Mar 26 12:25:33 hallowell agent[12907]: #011/omnibus/src/datadog-iot-agent/src/github.com/DataDog/datadog-agent/pkg/metrics/context_metrics.go:153 +0xe4
Mar 26 12:25:33 hallowell agent[12907]: github.com/DataDog/datadog-agent/pkg/metrics.(*ContextMetricsFlusher).FlushAndClear(0x4d417f0, 0x4d417fc, 0x0)
Mar 26 12:25:33 hallowell agent[12907]: #011/omnibus/src/datadog-iot-agent/src/github.com/DataDog/datadog-agent/pkg/metrics/context_metrics_flusher.go:55 +0x15c
Mar 26 12:25:33 hallowell agent[12907]: github.com/DataDog/datadog-agent/pkg/aggregator.(*TimeSampler).flushContextMetrics(0x4be4408, 0x4d417f0, 0x4d417fc)
Mar 26 12:25:33 hallowell agent[12907]: #011/omnibus/src/datadog-iot-agent/src/github.com/DataDog/datadog-agent/pkg/aggregator/time_sampler.go:212 +0x24
Mar 26 12:25:33 hallowell agent[12907]: github.com/DataDog/datadog-agent/pkg/aggregator.(*TimeSampler).flushSeries(0x4be4408, 0x623f3e7a, 0x0, 0x2044278, 0x4ef08a0)
Mar 26 12:25:33 hallowell agent[12907]: #011/omnibus/src/datadog-iot-agent/src/github.com/DataDog/datadog-agent/pkg/aggregator/time_sampler.go:128 +0x140
Mar 26 12:25:33 hallowell agent[12907]: github.com/DataDog/datadog-agent/pkg/aggregator.(*TimeSampler).flush(0x4be4408, 0x9f5331bf, 0x41d88fcf, 0x2044278, 0x4ef08a0, 0x80, 0xa6b27864, 0x286dc)
Mar 26 12:25:33 hallowell agent[12907]: #011/omnibus/src/datadog-iot-agent/src/github.com/DataDog/datadog-agent/pkg/aggregator/time_sampler.go:197 +0xa4
Mar 26 12:25:33 hallowell agent[12907]: github.com/DataDog/datadog-agent/pkg/aggregator.(*BufferedAggregator).getSeriesAndSketches(0x4be43c0, 0x51e04842, 0xc087ed7f, 0xe33ba52, 0x4, 0x2dd3c40, 0x2044278, 0x4ef08a0, 0x0, 0x0, ...)
Mar 26 12:25:33 hallowell agent[12907]: #011/omnibus/src/datadog-iot-agent/src/github.com/DataDog/datadog-agent/pkg/aggregator/aggregator.go:482 +0x120
Mar 26 12:25:33 hallowell agent[12907]: github.com/DataDog/datadog-agent/pkg/aggregator.(*BufferedAggregator).flushSeriesAndSketches(0x4be43c0, 0x51e04842, 0xc087ed7f, 0xe33ba52, 0x4, 0x2dd3c40, 0x4a8f000)
Mar 26 12:25:33 hallowell agent[12907]: #011/omnibus/src/datadog-iot-agent/src/github.com/DataDog/datadog-agent/pkg/aggregator/aggregator.go:649 +0x104
Mar 26 12:25:33 hallowell agent[12907]: github.com/DataDog/datadog-agent/pkg/aggregator.(*BufferedAggregator).Flush(0x4be43c0, 0x51e04842, 0xc087ed7f, 0xe33ba52, 0x4, 0x2dd3c40, 0x0)
Mar 26 12:25:33 hallowell agent[12907]: #011/omnibus/src/datadog-iot-agent/src/github.com/DataDog/datadog-agent/pkg/aggregator/aggregator.go:767 +0x74
Mar 26 12:25:33 hallowell agent[12907]: github.com/DataDog/datadog-agent/pkg/aggregator.(*BufferedAggregator).run(0x4be43c0)
Mar 26 12:25:33 hallowell agent[12907]: #011/omnibus/src/datadog-iot-agent/src/github.com/DataDog/datadog-agent/pkg/aggregator/aggregator.go:816 +0x35c
Mar 26 12:25:33 hallowell agent[12907]: github.com/DataDog/datadog-agent/pkg/aggregator.(*AgentDemultiplexer).Run(0x4cff980)
Mar 26 12:25:33 hallowell agent[12907]: #011/omnibus/src/datadog-iot-agent/src/github.com/DataDog/datadog-agent/pkg/aggregator/demultiplexer.go:248 +0x38
Mar 26 12:25:33 hallowell agent[12907]: created by github.com/DataDog/datadog-agent/pkg/aggregator.InitAndStartAgentDemultiplexer
Mar 26 12:25:33 hallowell agent[12907]: #011/omnibus/src/datadog-iot-agent/src/github.com/DataDog/datadog-agent/pkg/aggregator/demultiplexer.go:138 +0xc4
Mar 26 12:25:33 hallowell systemd[1]: datadog-agent.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Mar 26 12:25:33 hallowell systemd[1]: datadog-agent.service: Failed with result 'exit-code'.
```

**Comment by koukihai** [comment]
Downgrading to the previous version seems to work as a temporary workaround
In my case:
```
sudo apt install datadog-iot-agent=1:7.33.1-1
```

**Comment by jdiller** [comment]
Seems to have done the trick for me as well.

**Comment by djmitche** [comment]
This seems to be https://golang.org/pkg/sync/atomic/#pkg-note-BUG.
> On ARM, 386, and 32-bit MIPS, it is the caller's responsibility to arrange for 64-bit alignment of 64-bit words accessed atomically.

**Comment by mx-psi** [comment]
👋 Hi there, thanks for taking the time to report this! This is indeed due to the atomic issue @djmitche mentioned. We expect to fix this on the IoT Agent 7.36, until then we removed the 7.34 and 7.35 IoT Agent 32 bit ARM artifacts from our apt and yum repositories, so please use the latest released version.

As a side note, depending on your package manager cache state you may need to force an update (e.g. `apt-get update` or `yum check-update`) to pick up the changes from our repositories.
